src/SynthesiserRelation.cpp

```diff
@@ -0,0 +1,1729 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2018, The Souffle Developers. All rights reserved
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+#include "SynthesiserRelation.h"
+
```

Comment:
Include what you use - anything that isn't part of a method signature, since you can assume the header has that.